### ISSUE TITLE
Issue with assertVue since v6.10.0

### ISSUE THREAD

**Issue by lk77** [laravel/dusk#850]
- Dusk Version: 6.10.0
- Laravel Version: 8.19.0
- PHP Version: 7.3.24
- Database Driver & Version: mysql 8

### Description:

Hello,

My tests were passing and then i upgraded dusk :

```
Upgrading laravel/dusk (v6.9.1 => v6.10.0): Extracting archive
```

And my tests were not passing anymore :

```
 ErrorException: Array to string conversion
   │
   │ /vendor/laravel/dusk/src/Concerns/MakesAssertions.php:887
```

### Steps To Reproduce:

use an array as the value :

```
$browser->assertVue('value', [1,2,3], '@selector')
```

it seems that this line (887) of MakesAssertions cause the error :

```
"Did not see expected value [{$value}] at the key [{$key}]."
```

for me an array is a valid value in this case, and the vueAttribute method do return an array.

For now i rolled back to 6.9.1 and added <6.10.0 in my composer.json

thanks.

**Comment by driesvints** [comment]
An array is not support here: https://github.com/laravel/dusk/blob/6.x/src/Concerns/MakesAssertions.php#L878

**Comment by lk77** [comment]
But this makes no sens, phpunit supports it, there is no reason to not support arrays.

**Comment by driesvints** [comment]
@lk77 we had another issue about this here: https://github.com/laravel/dusk/issues/845

Like I said there: anyone is free to send in a PR if you want to add support for this.

**Comment by lk77** [comment]
I will make a Browser macro that restore the correct behaviour from 6.9.1 then

**Comment by lk77** [comment]
i'm replying here @driesvints, if you don't want to discuss the matter anymore i understand, but no need to lock someone else thread because of me

**Comment by UniquePanda** [comment]
Hi all and @driesvints ,

sorry to come back to this again, but I also just stumbled over it and found the two discussions here.  

**TL;DR: I think it is impossible to create a PR for this, considering what has been said in the discussions so far. How to allow array equality checks without making the `assertVue` method accept mixed?**

I would be willing to open a PR for this but after some thinking came to the conclusion, that right now, it seems to be impossible to get it accepted. I understand, that you have to be strict in many cases, but in this case it may be good to have a second thought about it. Let me explain:  

As others stated, PHPUnit would support values that are not strings for the needed assertions.  
Now dusk states in its DocBlock that only strings are allowed as values. There might be some considerations behind this, that we just don't know, so in general that's fine.  
The problem is, that the method is named `assertVue` which is a very general name and right now (I guess) many people use it to not just assert for string values, but also for arrays (which do not work anymore) and numbers (which still work).  
As the DocBlock implicitly says that numbers are also not valid, they only work because they are not explicity blocked off by the code.  

The previous discussion contained the statement "I don't think using mixed is a good solution."  
This means that simply reworking the DocBlock and exception message seems to be an approach that would not be accepted as a PR.  

Given the current name of the method, there seems to be no way to add the wanted functionality (allowance of comparison of other types than string) without having an inconsistent naming scheme.  
I could add methods like `assertVueArray`, `assertVueInteger` etc., but then the original method would have to be named `assertVueString` which obviously is not a good solution due to it being a breaking change.   
Maybe a method `assertVueTypeless` could be added but then no one would need to use the original method anymore and the original method would probably just be a wrapper that calls `assertVueTypeless` but blocks of every value that is not a string. Also not very clean.  

Therefore it seems to be impossible to create a PR right now that allows us to use this feature, which is actually quite simple to implement.  

Maybe we can get some information on why it is important that the current method only takes strings?  
Maybe it was just never intended to test for anything else? Then I would know that I have to create my own methods outside of dusk. However then the answer "Please create a PR for that" is not very helpful :D  
Or would it just be sufficient to allow all types that might make sense and block off any other types (so that the error message can be fixed accordingly)? E.g. explicitly allowing strings, ints, floats, arrays, ...?

**Comment by driesvints** [comment]
> Maybe we can get some information on why it is important that the current method only takes strings?

It's not important, it's just the way it works right now. It can perfectly be adjusted to something like this:

```php
    /**
     * Assert that the Vue component's attribute at the given key has the given value.
     *
     * @param  string  $key
     * @param  string|array  $value
     * @param  string|null  $componentSelector
     * @return $this
     */
    public function assertVue($key, $value, $componentSelector = null)
    {
        $message = implode(',', (array) $value);

        PHPUnit::assertEquals(
            $value,
            $this->vueAttribute($componentSelector, $key),
            "Did not see expected value [{$message}] at the key [{$key}]."
        );

        return $this;
    }
```

Which I just tried to do. However I don't know too much of Vue to know how to adjust the test suite for this etc. 

Anyone can send in a PR for this. We don't need to add methodX, methodY.
